src/Features/Core/Portable/Completion/CompletionHelper.cs

```diff
@@ -163,19 +163,15 @@ public bool MatchesPattern(CompletionItem item, string pattern, CultureInfo cult
         private PatternMatcher GetPatternMatcher(string pattern, CultureInfo culture, bool includeMatchedSpans)
             => GetPatternMatcher(pattern, culture, includeMatchedSpans, _patternMatcherMap);
 
-        public int CompareItems(CompletionItem item1, PatternMatch? match1, CompletionItem item2, PatternMatch? match2, out bool onlyDifferInCaseSensitivity)
+        public int CompareItems(CompletionItem item1, PatternMatch? match1, CompletionItem item2, PatternMatch? match2, bool filterTextIsLowerCaseOnly)
```

Comment:
this is the part that confuses me.  i don't see why it matters if the filter-text is lowercase only.  i would expect that even if the filter-text is cased in any fashion, that an exact/prefix match for it must always be better than all other comparison points.

Reply:
holding off on going further until clarification on this part.

Reply:
Fixed to not prefer item of `MatchPriority.Preselect` over case-sensitive matches.